src/thing_list.c

```diff
@@ -3514,7 +3514,48 @@ struct Thing *get_creature_of_model_training_at_subtile_and_owned_by(MapSubtlCoo
 
 struct Thing *get_nearest_object_at_position(MapSubtlCoord stl_x, MapSubtlCoord stl_y)
 {
-  return _DK_get_nearest_object_at_position(stl_x, stl_y);
+  // return _DK_get_nearest_object_at_position(stl_x, stl_y);
+  long OldDistance = 0x7FFFFFFF;
+  struct Thing *thing;
+  long NewDistance;
+  signed int n;
+  signed int k = 0;
+  struct Thing *result = NULL;
+  MapSubtlCoord x, y;
+  struct Map *blk;
+  do
+  {
+    if ( stl_y + k >= 0 && stl_y + k < 256 )
+    {
+      n = 0;
+      y = stl_y + k;
+      do
+      {
+        if ( stl_x + n >= 0 && stl_x + n < 256 )
+        {
+          x = stl_x + n;
+          blk = get_map_block_at(x, y);
+          for ( thing = thing_get(get_mapwho_thing_index(blk)); (!thing_is_invalid(thing)); thing = thing_get(thing->next_on_mapblk) )
+          {
+            if (thing->class_id == TCls_Object)
+            {
+                NewDistance = get_2d_box_distance_xy(stl_x, stl_y, thing->mappos.x.stl.num, thing->mappos.y.stl.num);
+                if ( NewDistance < OldDistance )
+                {
+                    OldDistance = NewDistance;
+                    result = thing;
+                }
+            }
+          }
+        }
+        n++;
+      }
+      while ( n < STL_PER_SLB );
+    }
+    k++;
+  }
+  while ( k < STL_PER_SLB );
+  return result;
```

Comment:
This function is named wrong. 
It is counting subtiles from 0 to 2.
So it cant find an object 1 subtile left from given stl_x, stl_y

Reply:
I think that's how it is in the original code.

Reply:
Looking into this, I think the function is only used to find hero gates and lair totems. Heroes gates are appear to be about a slab tall(so 3 STL), and when facing north (and precision sell helps test this) your mouse over both the exact subtile, but also 2 subtiles north and/or 1 subtile left or right will give you the tooltip.

But for lair totems I do not understand. 
Here it seems to match the behavior sim describes:

![image](https://user-images.githubusercontent.com/13840686/115939699-3459fc80-a49f-11eb-9e7c-1cd162b56787.png)

With Adams version it picks up the lair when I put my cursor left from it, not when I put it right from it. However, with the master it picks it up perfectly. (Note, only some units get a tooltip, a wizard does, a knight does not.)

Reply:
You mean it's supposed to pick up the object when the cursor is just to its right too?

Reply:
In the master, it picks it up only when you hold it over. I think you want to replicate the behavior of the master.